### PR TITLE
D.R.Y up Vim modeline expression

### PR DESCRIPTION
**Pull request by Alhadis**
## Description
Fixes:#5347 by updating the Vim Help heuristic with fixes made in #5271 (which should've included `heuristics.yml` as well… 🤦). I've added a reminder to `modeline.rb` so something like this doesn't happen again.

*(Template removed as it doesn't apply)*

### CODE REVIEW

**Comment by lildude** [comment]
I think we also need to add `help` as an alias for "Vim Help" in order to fix https://github.com/github/linguist/issues/5347

**Comment by Alhadis** [comment]
Derp. 🤦🤦 You're right.

At least one of us is paying attention. 😅

**Comment by Alhadis** [comment]
@lildude We've hit a snag:

~~~
  2) Failure:
TestBlob#test_language [/home/runner/work/linguist/linguist/test/test_blob.rb:266]:
textobj-rubyblock.vba.
Expected: "Vim script"
  Actual: "Vim Help File"
~~~

This is failing because a VimBall file (see #4412) ends with a help file modeline:

https://github.com/github/linguist/blob/19b3a9bd7d585be8d6459cfab0a51607b2aee68b/samples/Vim%20script/textobj-rubyblock.vba#L215

As the name suggests, VimBalls are the Vim equivalent of a tarball: a concatenation of several files with headers betwixt each blob of data:

https://github.com/github/linguist/blob/19b3a9bd7d585be8d6459cfab0a51607b2aee68b/samples/Vim%20script/textobj-rubyblock.vba#L62-L66

I'm not sure what to do here.

**Comment by lildude** [comment]
Oh (vim)balls!!! 

> I'm not sure what to do here.

Yeah, I'm not too sure either. That change is needed in order to fix https://github.com/github/linguist/issues/5347 (cos of the size of the file) but doing so breaks VimBall detection.  I'll  need to have a think about it.

**Comment by lildude** [comment]
Got it!! Vimballs are really easy to find thanks to the first few lines always containing `UseVimball`.

We can use this to our advantage in the modeline strategy with no extra cost. 

I'll push my changes to this branch.

File: lib/linguist/strategy/modeline.rb [hunk 1]
```diff
@@ -144,6 +145,9 @@ def self.call(blob, _ = nil)
         return [] if blob.symlink?
 
         header = blob.first_lines(SEARCH_SCOPE).join("\n")
+        # Return early for Vimball files as their modeline will not reflect their filetype.
+        return [] if header.include?("UseVimball")
+
```
Comment (lildude) [inline lib/linguist/strategy/modeline.rb:150]: What do you think of this @Alhadis?

Comment (Alhadis) [reply lib/linguist/strategy/modeline.rb:150]: Looks great! (I was about to suggest that a regression test be added, but remembered that most samples are implicitly used as fixtures anyway... 😅)

**Review by lildude — APPROVED** [review]

**Pull request merged**